pkg/api/handlers/generic/containers.go

```diff
@@ -0,0 +1,306 @@
+package generic
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"github.com/containers/libpod/pkg/api/handlers/utils"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/containers/libpod/libpod"
+	"github.com/containers/libpod/libpod/define"
+	"github.com/containers/libpod/libpod/logs"
+	"github.com/containers/libpod/pkg/api/handlers"
+	"github.com/containers/libpod/pkg/util"
+	"github.com/docker/docker/api/types"
+	"github.com/gorilla/mux"
+	"github.com/gorilla/schema"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+func RemoveContainer(w http.ResponseWriter, r *http.Request) {
+	decoder := r.Context().Value("decoder").(*schema.Decoder)
+	query := struct {
+		Force bool `schema:"force"`
+		Vols  bool `schema:"v"`
+		Link  bool `schema:"link"`
+	}{
+		// override any golang type defaults
+	}
+
+	if err := decoder.Decode(&query, r.URL.Query()); err != nil {
+		handlers.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest,
+			errors.Wrapf(err, "Failed to parse parameters for %s", r.URL.String()))
+		return
+	}
+	if query.Link {
+		handlers.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest,
+			errors.New("link is not supported"))
+		return
+	}
+	utils.RemoveContainer(w, r, query.Force, query.Vols)
+}
+
+func ListContainers(w http.ResponseWriter, r *http.Request) {
+	runtime := r.Context().Value("runtime").(*libpod.Runtime)
+
+	containers, err := runtime.GetAllContainers()
+	if err != nil {
+		handlers.InternalServerError(w, err)
+		return
+	}
+
+	infoData, err := runtime.Info()
+	if err != nil {
+		handlers.InternalServerError(w, errors.Wrapf(err, "Failed to obtain system info"))
```

Comment:
No capitalized errors.

Reply:
to be consistent with the other api, we must use caps.